CIP-0086 | Remove metadata update transaction restrictions (#521)
cardano-foundation/CIPs: CIP-0086/README.md

### Simple metadata updates

> A simple metadata update transaction may contain any inputs and outputs,
> as these are ignored for the purposes of updating token metadata.

Reviewer: ```suggestion
A simple metadata update transaction must be signed
by the signing key of the oracle update address,
but it may contain any inputs and outputs,
as these are ignored for the purposes of updating token metadata.
```

}
```

> A tabular metadata update transaction may contain any inputs and outputs,
> as these are ignored for the purposes of updating token metadata.

Reviewer: ```suggestion
A tabular metadata update transaction must be signed
by the signing key of the oracle update address,
but it may contain any inputs and outputs,
as these are ignored for the purposes of updating token metadata.
```

scan through the sequence of transactions in a Cardano node’s blockchain,
applying the CIP-25 and CIP-86 updates
in the order that they are encountered in this sequence.
Should a transaction contain both CIP-25 and CIP-86 updates,
then the CIP-25 updates should be applied first, 
followed by the CIP-86 updates.

> A transaction that contains a CIP-25 update may also
> contain an explicit oracle assignment,
> but these can be applied in parallel because they do not clash with each other.

Reviewer: ```suggestion
If a transaction contains both a CIP-25 update
and a CIP-86 explicit oracle assignment,
then the CIP-25 update will be applied as usual
and the oracle addresses (main and/or update) will be set to
those of the explicit CIP-86 oracle assignment.
```

cached in the indexer database for the respective asset classes.

For each oracle main address currently assigned to a policy ID,

> the indexer monitors the blockchain for transactions
> that contain oracle assignment updates in their metadata.

Reviewer: ```suggestion
the indexer monitors the blockchain for transactions
that contain oracle assignment updates in their metadata
and are signed by the signing key of the main address.
```

The indexer updates its database to reflect these explicit oracle assignments
and removes any implicit assignments that were replaced by explicit assignments.

For each oracle update address currently assigned to a policy ID,

> the indexer monitors the blockchain for transactions
> that contain CIP-86 token metadata updates.

Reviewer: ```suggestion
the indexer monitors the blockchain for transactions
that contain CIP-86 token metadata updates
and are signed by the signing key of the update address.
```

than to keep track of all the people who control various authorization tokens,
at a given time.

### Removing the original CIP-86 update transaction restriction

In the first draft of this proposal,
we prohibited CIP-86 oracle assignment updates and metadata updates
from occurring in transactions that mint tokens,
in order to avoid awkward clashes with CIP-25 metadata transactions.
This was removed because, while it likely does not make sense 
to create CIP-25 and CIP-86 metadata for the same token in one transaction,
it could feasibly make sense to want to update the metadata 
for other tokens while minting another one.

We also originally required CIP-86 updates to occur in transactions
that only send ADA from an oracle address (main or update, as appropriate),
to prevent unforeseen interactions with other mechanisms
that may have negative consequences.

> This requirement was removed for a two reasons.

Reviewer: ```suggestion
This requirement was removed for two reasons.
```

than to keep track of all the people who control various authorization tokens,
at a given time.

> ### Restricting CIP-86 update transactions

Reviewer 2: @nalane instead of simply striking out this section, can you leave the explanation of why the "non-minting transaction" requirement was originally dictated, but then (equally importantly) explain why that requirement was dropped?

Author: Hi @rphair, I have added the section as requested

Reviewer 2: thanks @nalane (resolved with https://github.com/cardano-foundation/CIPs/pull/521/commits/c7583bc08a844249db634e8930a14de3034db42d)
